ast/capabilities.go

```diff
@@ -63,3 +66,39 @@ func LoadCapabilitiesJSON(r io.Reader) (*Capabilities, error) {
 	var c Capabilities
 	return &c, d.Decode(&c)
 }
+
+// LoadCapabilitiesVersionJSON loads a JSON serialized capabilities structure from the the specific version.
+func LoadCapabilitiesVersionJSON(version string) (*Capabilities, error) {
```

Comment:
[nit] is "...JSON" actually meaningful here? We're returning the unmarshalled struct, after all. The fact that it's JSON in the embed.FS doesn't really matter much, I think 🤔 WDYT?

Reply:
Cool idea. But removing the JSON suffix there, would also mean that the suffix on **__** should also be removed, because we would have two functions, which return the same thing, but would look like this:
```
// LoadCapabilitiesJSON loads a JSON serialized capabilities structure from the reader r.
func LoadCapabilitiesJSON(r io.Reader) (*Capabilities, error) {}

// LoadCapabilitiesVersion loads a JSON serialized capabilities structure from the specific version.
func LoadCapabilitiesVersion(version string) (*Capabilities, error) {}
```
WDYT? :smiley:

Reply:
> But, removing the JSON suffix there, would also mean that the suffix on the already existing function `LoadCapabilitiesJSON` should also be removed, because we would have two functions, which return the same thing `(*Capabilities, error)`

Hmm that function, however, expects an `io.Reader` providing capabilities json content and will unmarshal the result accordingly. (It's also exported and thus shouldn't be changed without a very good reason 😅)

I guess we could do this, pseudo-code:

```go
LoadCapabilitiesVersion(v string) (*Capabilities, error) {
  r := ... // get an io.Reader from the embed.FS data for this version (if it exists)
  return LoadCapabilitiesJSON(r) // existing function
}
```

I suppose that's not entirely unlike what you've described, though, is it? What do you think?

Reply:
2373e6ff7aac875ce5e0b522144bcd1826c31914 I agree!